aten/src/ATen/functorch/ADInterpreters.cpp

```diff
@@ -136,12 +156,29 @@ static void autogradBasedTransformSendToNext(
   if (getDynamicLayerStack().size() == 0) {
     sanityCheckStack(op, stack);
   }
-  op.callBoxed(stack);
 
   // Step 4, 5, 6
   auto ret_size = op.schema().returns().size();
+
+  op.callBoxed(stack);
+
+  auto aliased_input_outputs = findAliasedInputs(op.schema());
+  std::vector<int64_t> unwrapped_outputs; // indexes of outputs that should remain unwrapped
+
+  // lift_fresh: it's must be freshly allocated and should be wrapped. User shouldn't have access to input version
+  // alias: this is needed for the CompositeImplicit instance norm (running_mean/var get set to be a wrapped value)
+  //        It's not a user facing function, but is more prone to possible errors
+  const bool always_wrap = op.schema().name() == "aten::lift_fresh" || op.schema().name() == "aten::alias";
```

Comment:
String comparison is slow

Reply:
Locally I'm seeing that with 10M tensor creations it goes from ~59s with this check to ~54s without (replacing this with a constant true). I upped it to try to get rid of too much noise but basically having this is ~10% slowdown per call

Reply:
We can do something like what we previously did with the JVP decomposition. There, we also needed to do string comparison, but I was worried it would be too slow, so we added a workaround. The workaround was something like:
- ATenOpId is an enum with various operators
- DynamicLayerFront gets a registration for each operator that we care about (in this case, alias and lift_fresh). These kernels will pass the correct ATenOpId down the line
- We can give each Interpreter::process a Interpreter::process_with_id that takes an additional ATenOpId and does something with it


Code for reference:
https://github.com/pytorch/pytorch/commit/f10e9d66c49c023bdf1a819a10b327daaeb161c5#diff-edf34e92e4e457dea93693fc196c7a6af5ec14d7e9a2f9aa51529d6d7427517aR369

Long-term it sounds like we need some sort of string interning for dispatcher operations. Let me write up an issue on it. It might be doable to write that in a couple of days and that would be better than if we cooked up the hack above

Reply:
Updated! I made it more simple and specialized (sendToNext now just accepts an extra bool) than the forward over reverse example since both lift_fresh and alias have the same behavior. There's a note there though that if we need more special behavior than just this, it's going to scale better to use enums

Reply:
Approach SGTM, will read the code closer soon. Do you also have microbenchmarks for this new approach?

Reply:
Small enough to probably just be noise (for 10M calls)
Without this check: 54.707358837127686s
With: 54.98072600364685s

Reply:
That's awesome